tests/playwright/src/model/pages/pull-image-page.ts

```diff
@@ -37,17 +40,73 @@ export class PullImagePage extends BasePage {
     this.backToImagesLink = page.getByRole('link', { name: 'Go back to Images' });
     this.manageRegistriesButton = page.getByRole('button', { name: 'Manage registries' });
     this.imageNameInput = page.getByLabel('Image to Pull');
+    this.tabContent = page.getByRole('region', { name: 'Tab Content', exact: true });
+    this.searchResultsTable = this.tabContent.getByRole('row');
+    this.doneButton = page.getByRole('button', { name: 'Done', exact: true });
   }
 
-  async pullImage(imageName: string, tag = '', timeout = 60000): Promise<ImagesPage> {
-    const fullImageName = `${imageName}${tag.length === 0 ? '' : ':' + tag}`;
-    await this.imageNameInput.fill(fullImageName);
-    await playExpect(this.pullImageButton).toBeEnabled();
-    await this.pullImageButton.click();
+  async pullImage(imageName: string, tag = '', timeout = 60_000): Promise<ImagesPage> {
+    return test.step(`Pulling image ${imageName}:${tag}`, async () => {
+      const fullImageName = `${imageName}${tag.length === 0 ? '' : ':' + tag}`;
+      await this.imageNameInput.fill(fullImageName);
+      await playExpect(this.pullImageButton).toBeEnabled();
+      await this.pullImageButton.click();
 
-    const doneButton = this.page.getByRole('button', { name: 'Done' });
-    await playExpect(doneButton).toBeEnabled({ timeout: timeout });
-    await doneButton.click();
-    return new ImagesPage(this.page);
+      await playExpect(this.doneButton).toBeEnabled({ timeout: timeout });
+      await this.doneButton.click();
+      return new ImagesPage(this.page);
+    });
+  }
+
+  async getAllSearchResultsFor(imageName: string, searchForVersion: boolean, imageTag = ''): Promise<string[]> {
```

Comment:
It would be good to add a function that clears out the selection.

Reply:
I mean that's just clearing out the input element, it doesn't really require a function.

Reply:
added a method to clear the search box.